applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp

```diff
@@ -904,21 +904,10 @@ namespace Kratos {
 
         ConditionsArrayType& pConditions = GetFemModelPart().GetCommunicator().LocalMesh().Conditions();
         const ProcessInfo& r_process_info = GetFemModelPart().GetProcessInfo();
-        Vector rhs_cond;
-        std::vector<unsigned int> condition_partition;
-        OpenMPUtils::CreatePartition(mNumberOfThreads, pConditions.size(), condition_partition);
-
-        #pragma omp parallel for private (rhs_cond)
-        for (int k = 0; k < mNumberOfThreads; k++) {
-            ConditionsArrayType::iterator it_begin = pConditions.ptr_begin() + condition_partition[k];
-            ConditionsArrayType::iterator it_end = pConditions.ptr_begin() + condition_partition[k + 1];
-
-            for (ConditionsArrayType::iterator it = it_begin; it != it_end; ++it) {
-                it->FinalizeSolutionStep(r_process_info);
-            }
-        }
 
+        block_for_each(pConditions, [&r_process_info](ModelPart::ConditionType& rCondition){
```

Comment:
do we need just `[&r_process_info]` here or can we use the general `[&]` ?

Reply:
I would use the least possible, i.e. `[&r_process_info]`
can help to prevent some hidden issues

Reply:
Both options are ok, but according to @philbucher in #https://github.com/KratosMultiphysics/Kratos/pull/8614#discussion_r614710895 it is preferred to pass only the needed ones.

Reply:
Ok, I am still learning the basics about lambda functions. Sorry about the noise.